tests/extras/test_vcf_annotation.py

```diff
@@ -0,0 +1,40 @@
+"""Ensure proper functionality of VCFAnnotator"""
+import gzip
+import os
+
+import pytest
+
+from ga4gh.vrs.extras.vcf_annotation import VCFAnnotator
+from ga4gh.vrs.extras.translator import Translator
+
+
+@pytest.fixture(scope="module")
+def tlr_local(dataproxy):
+    return Translator(data_proxy=dataproxy)
+
+
+@pytest.fixture(scope="module")
+def vcf_annotator(tlr_local):
+    return VCFAnnotator(tlr_local)
+
+
+@pytest.mark.vcr
+def test_annotate_vcf(vcf_annotator):
+    TEST_DATA_DIR = "tests/extras/data"
+
+    input_vcf = f"{TEST_DATA_DIR}/test_vcf_input.vcf"
+    output_vcf = f"{TEST_DATA_DIR}/test_vcf_output.vcf.gz"
+    output_vrs_pkl = f"{TEST_DATA_DIR}/test_vcf_pkl.pkl"
+    expected_vcf = f"{TEST_DATA_DIR}/test_vcf_expected_output.vcf.gz"
+    vcf_annotator.annotate(input_vcf, output_vcf, output_vrs_pkl)
+
+    with gzip.open(output_vcf, "rt") as out_vcf:
+        out_vcf_lines = out_vcf.readlines()
+    with gzip.open(expected_vcf, "rt") as expected_output:
+        expected_output_lines = expected_output.readlines()
+
+    assert out_vcf_lines == expected_output_lines
+    assert os.path.exists(output_vrs_pkl)
+
+    os.remove(output_vcf)
+    os.remove(output_vrs_pkl)
```

Comment:
I think these cleanup calls won't run if the tests fail -- the typical pattern is to use fixtures and `yield` statements to enable "teardown" behavior which is explained here: https://stackoverflow.com/a/62784688. I can walk through it if this is unclear given that I know we want to get this PR merged

(another maybe simpler way is to use tempfiles -- https://docs.python.org/3/library/tempfile.html -- which will autodelete)